Purchases/Purchasing/StoreKit2/StoreKit2TransactionListener.swift

```diff
@@ -44,23 +48,25 @@ class StoreKit2TransactionListener {
         self.taskHandle = nil
     }
 
-    func handle(purchaseResult: StoreKit.Product.PurchaseResult) async -> Bool {
+    /// - Returns: whether the user cancelled
+    /// - Throws: Error if purchase was not completed successfully
+    func handle(purchaseResult: StoreKit.Product.PurchaseResult) async throws -> Bool {
         switch purchaseResult {
         case .success(let verificationResult):
-            await handle(transactionResult: verificationResult)
-            // todo: proper handling
-            return false
+            if try await handle(transactionResult: verificationResult) {
+                return false
+            } else {
+                throw ErrorCode.purchaseInvalidError
```

Comment:
This would only happen on an unverified purchase, right? in `handle(transactionResult:` we have a `storeProblemError` for this case, and here we do `.purchaseInvalidError`, we should settle on one

Reply:
Yup removed this since `handle` never returned `false`.